packages/amplify_authenticator/example/integration_test/utils/mock_data.dart

```diff
@@ -34,10 +34,54 @@ String generatePassword() =>
     uppercaseLetters[random.nextInt(uppercaseLetters.length)] +
     symbols[random.nextInt(symbols.length)];
 
-String generatePhone() => '+155501${randomPhoneDigits()}';
+/// generates a fake US phone number using a reserved number in the North America
+///
+/// See https://en.wikipedia.org/wiki/Fictitious_telephone_number#North_American_Numbering_Plan
+_PhoneNumber generateUSPhoneNumber() => _PhoneNumber(
+      countryCode: '+1',
+      areaCode: '555',
+      phoneNumber: '01${randomPhoneDigits(2)}',
+    );
+
+/// generates a fake french phone number using a reserved number in france
+///
+/// See https://en.wikipedia.org/wiki/Fictitious_telephone_number#France
+_PhoneNumber generateFrenchPhoneNumber() => _PhoneNumber(
+      countryCode: '+33',
+      areaCode: '1',
+      phoneNumber: '9900${randomPhoneDigits(4)}',
+    );
 
 String generateUsername() => 'flutter-user-${randomNumber()}';
 
 int randomNumber() => random.nextInt(1 << 30);
 
-String randomPhoneDigits() => '${random.nextInt(10)}${random.nextInt(10)}';
+String randomPhoneDigits(int count) {
+  String value = '';
+  for (var i = 0; i < count; i++) {
+    value += random.nextInt(10).toString();
+  }
+  return value;
+}
+
+class _PhoneNumber {
```

Comment:
Seems like this might be a useful class in Authenticator, instead of working with strings

Reply:
Agreed. I am a little hesitant to do this right now though. I started to, but it got messy quickly. I think we need to do some (not all) of the refactoring work around AuthenticatorFormField I proposed in https://github.com/aws-amplify/amplify-flutter/pull/1116. This would make the form field code easier to understand. In particular, the current definition of AuthenticatorFormField I think can be simplified.

The current definition is: 

```dart
abstract class AuthenticatorFormField<FieldType, FieldValue,
        T extends AuthenticatorFormField<FieldType, FieldValue, T>>
    extends AuthenticatorComponent<T> {
}
```

IMO it is hard to follow what FieldType, FieldValue, and T are in this context. I am pretty sure we could simplify it to something like:

```dart
abstract class AuthenticatorFormField<FieldType>
    extends AuthenticatorComponent {
}
```

In the above definition, FieldType would be the data type for the data in the form field (String, PhoneNumber, etc.). In the previous refactor PR I put up, I simplified it further remove both `FieldType` and `FieldValue`. While that worked, it meant that every field was a String, which isn't what we want.